test/extended/image_ecosystem/scl.go

```diff
@@ -106,6 +106,13 @@ var _ = g.Describe("[image_ecosystem][Slow] openshift images should be SCL enabl
 		o.Expect(err).NotTo(o.HaveOccurred())
 	})
 
+	g.AfterEach(func() {
+		if g.CurrentGinkgoTestDescription().Failed {
+			exutil.DumpPodStates(oc)
+			exutil.DumpPodLogsStartingWith("", oc)
```

Comment:
you sure you want to dump all pod logs?  Just the ones in the namespace maybe?  @stevekuznetsov I thought we already collected all pod logs.  @sjenning since I had an open question on "does the kubelet delete our logs" and this seems to suggest that at least some are missing.

Reply:
Contrary to the function name, this is just the ones in the oc client's current namespace (https://github.com/openshift/origin/blob/master/test/extended/util/framework.go#L193).  i'm doing this because the "collect all the pod logs" has proven unreliable(we are definitely not getting all the container logs today in the s3 artifacts) and i don't want to fight that battle anymore.

Reply:
(note that garbage collection or other factors can also break collecting all the container logs at the end of the extended run, so grabbing logs as soon as the test fails is more likely to be successful)

Reply:
> (note that garbage collection or other factors can also break collecting all the container logs at the end of the extended run, so grabbing logs as soon as the test fails is more likely to be successful)

This was a question I explicitly asked @sjenning .  He said that the container logs were only cleaned when disk space became a problem.

Reply:
We get pod logs _after_ the e2e is finished executing. IIRC it cleans up test namespaces itself so we won't get those after an e2e test is finished.

Reply:
even if the namespaces are cleaned up the containers should still be around, right?  or are containers cleaned up when the pod is deleted?

Reply:
the latter